tests/Feature/Position/PositionTest.php

```diff
@@ -0,0 +1,174 @@
+                    'name' => $positions->first()->name,
+                    'frequency' => $positions->first()->frequency,
+                ]
+            ]
+        ]);
+    }
+
+    /* @test */
+    public function testItReturnsPositionByCallsign()
+    {
+        $positions = factory(Position::class, 2)->create();
+
+        $this->graphQL("
+        {
+            position(callsign: \"{$positions->first()->callsign}\") {
+                id
+                name
+                frequency
+            }
+        }
+        ")->assertJson([
+            'data' => [
+                'position' => [
+                    'id' => $positions->first()->id,
+                    'name' => $positions->first()->name,
+                    'frequency' => $positions->first()->frequency,
+                ]
+            ]
+        ]);
+    }
+
+    /** @test */
+    public function testValidPositionCanBeCreated()
+    {
+        $this->graphQL('
+          mutation { 
+            createPosition(name: "Bristol Tower", callsign: "EGGD_TWR", frequency: "133.850", type: 4) {
+                id
+                name
+            }
+          }')->assertJsonStructure([
+            'data' => [
+                'createPosition' => [
+                    'id',
+                    'name'
+                ]
+            ]
+        ]);
+
+        $this->assertDatabaseHas('positions', [
+            'name' => 'Bristol Tower',
+            'callsign' => 'EGGD_TWR',
+            'frequency' => '133.850'
+        ]);
+    }
+
+    /** @test */
+    public function testDuplicatePositionCannotBeCreated()
+    {
+        factory(Position::class)->create([
+            'name' => 'Bristol Tower',
+            'callsign' => 'EGGD_TWR',
+            'frequency' => '133.850'
+        ]);
+
+        $this->graphQL('
+            mutation { 
+                createPosition(name: "Bristol Tower", callsign: "EGGD_TWR", frequency: "133.850", type: 4) {
+                    id
+                    name
+                }
+            }')->assertJsonFragment(["validation" => [
+                "callsign" => [
+                    "The callsign has already been taken."
+                ]
+            ]]);
+    }
+
+    /** @test */
+    public function testItCanUpdateAPositon()
+    {
+        factory(Position::class)->create([
+            'id' => 1,
+            'name' => 'Bristol Tower',
+            'callsign' => 'EGGD_TWR',
+            'frequency' => '133.850'
+        ]);
+
+        $this->graphQL('
+            mutation { 
+                updatePosition(id: 1, name: "Slough Tower", callsign: "EGLL_TWR") {
+                    id
+                    name
+                }
+            }')->assertJson([
+                'data' => [
+                    'updatePosition' => [
+                        'id' => 1,
+                        'name' => 'Slough Tower'
+                    ]
+                ]
+        ]);
+
+        $this->assertDatabaseHas('positions', [
```

Comment:
I may be being picky... Do we want to assert that the database does not have the original one (i.e. Bristol Tower)? Thus validating that has in fact been updated, rather than just created?

Reply:
Yep! Good call.